neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -53,48 +54,103 @@ protected override void OnReceive(object message)
             switch (msg.Command)
             {
                 case "addr":
-                    OnAddrMessageReceived(msg.Payload.AsSerializable<AddrPayload>());
-                    break;
+                    {
+                        var payload = msg.Payload.AsSerializable<AddrPayload>();
+                        if (!Plugin.ReceivedMessageAllowed(msg.Command, payload)) return;
```

Comment:
Why not just pass the `msg` object to the plugins?

Reply:
We need the object deserialized for filters. The other implementation could be this, but the plugin must deserialize again the object, any idea?

Reply:
I think that deserialized is enough, otherwise maybe we need to expose some additional functions the plugins.